### DISCUSSION TITLE
Text area or editable text in the canvas

### DISCUSSION THREAD

**Discussion by komitoff** [konvajs/konva#1867] [General]
Hi everybody, 
I enjoy using the konvaJS library. It is amazing how much it can be achieved with it. 
But I am struggling with a basic functionality that I want to implement in my app that I am doing using the library - text editing. 
I have a canvas with some images and I want to leave notes for some of them. The note is just a rectangle with text, but on dblclick I want to have an edit mode. 
Currently I am using the way of attaching an html textarea in the exact same coordinates of the page with the same width and height as the rectangle, but sometimes this is tricky, because the main container  can be scaled or have some paddings. 
Is there a native way of having editable text-area of something similar in the canvas or better approach than  the one I am using currently. 

Thanks in advance!

**Comment by lavrton** [comment]
We don't have a native approach right now and not plans to support it as part of the library. 

I think these solutions works just fine:
https://konvajs.org/docs/sandbox/Editable_Text.html
https://konvajs.org/docs/sandbox/Rich_Text.html
